web/src/OverviewActionBar.tsx

```diff
@@ -84,41 +186,82 @@ type FilterRadioButtonProps = {
   filterSet: FilterSet
 }
 
-function FilterRadioButton(props: FilterRadioButtonProps) {
-  let level = props.level
-  let text = "All Logs"
+export function FilterRadioButton(props: FilterRadioButtonProps) {
+  let { level, filterSet } = props
+  let leftText = "All Levels"
   if (level === FilterLevel.warn) {
-    text = "Warnings"
+    leftText = "Warnings"
   } else if (level === FilterLevel.error) {
-    text = "Errors"
+    leftText = "Errors"
   }
+
   let isEnabled = level === props.filterSet.level
+  let rightText = (
+    <ExpandMoreIcon
+      style={{ width: "16px", height: "16px" }}
+      key="right-text"
+    />
+  )
+  let rightStyle = { paddingLeft: "4px", paddingRight: "4px" } as any
+  if (isEnabled) {
+    if (filterSet.source == FilterSource.build) {
+      rightText = <span key="right-text">Build</span>
+      rightStyle = null
+    } else if (filterSet.source == FilterSource.runtime) {
+      rightText = <span key="right-text">Runtime</span>
+      rightStyle = null
+    }
+  }
 
   // isRadio indicates that clicking the button again won't turn it off,
   // behaving like a radio button.
-  let className = "isRadio"
+  let leftClassName = "isRadio"
+  let rightClassName = ""
   if (isEnabled) {
-    className += " isEnabled"
+    leftClassName += " isEnabled"
+    rightClassName += " isEnabled"
   }
 
   let history = useHistory()
   let l = history.location
   let onClick = () => {
     let search = new URLSearchParams(l.search)
     search.set("level", level)
+    search.set("source", "")
     history.push({
       pathname: l.pathname,
       search: search.toString(),
     })
   }
+
+  let rightPillRef = useRef(null as any)
+  let [sourceMenuAnchor, setSourceMenuAnchor] = useState(null)
+  let onMenuOpen = (e: any) => {
+    setSourceMenuAnchor(e.currentTarget)
+  }
+  let sourceMenuOpen = !!sourceMenuAnchor
+
   return (
-    <ButtonRoot
-      style={{ marginRight: "16px" }}
-      className={className}
-      onClick={onClick}
-    >
-      {text}
-    </ButtonRoot>
+    <ButtonPill style={{ marginRight: "16px" }}>
```

Comment:
we'll eventually want a standard component library, and a good rule of thumb with this kind of spacing rule is that it shouldn't be part of the core component (since the spacing amount is about relations between components rather than the component itself)